src/conn/conn_api.c

```diff
@@ -190,6 +190,42 @@ __wt_conn_remove_collator(WT_SESSION_IMPL *session)
 }
 
 /*
+ * __conn_compat_config --
+ *	Configure compatibility version.
+ */
+static int
+__conn_compat_config(WT_SESSION_IMPL *session, const char **cfg)
+{
+	WT_CONFIG_ITEM cval;
+	WT_CONNECTION_IMPL *conn;
+
+	conn = S2C(session);
+	WT_RET(__wt_config_gets(session, cfg,
+	    "compatibility.major", &cval));
+	if (cval.val) {
+		if (cval.val <= WIREDTIGER_VERSION_MAJOR)
+			conn->compat_major = (uint16_t)cval.val;
+		else
+			WT_RET_MSG(session, EINVAL, "unknown major version");
+	} else
+		conn->compat_major = WIREDTIGER_VERSION_MAJOR;
+	WT_RET(__wt_config_gets(session, cfg,
+	    "compatibility.minor", &cval));
+	/*
+	 * We don't currently have knowledge of major/minor pairs.  If the
+	 * user gave us zero set it to the current version.  Check for some
+	 * out of bounds settings.
+	 */
+	conn->compat_minor = (uint16_t)cval.val;
```

Comment:
We can discuss this.  I chose this path (and wrote it up in the scope) for a couple of reasons:
* There already exists `version=(major=,minor=)` in the `api_data.py` file.  This API followed that precedent.
* This API allows a user to not have to know the specific release necessarily.  If this change goes into a 3.0, then just `compat=(major=2)` is sufficient.
* Given those reasons above and that it is easier to deal with integer args instead of bursting apart a string led to this API.

However, these are not strong reasons.  The `version` string only appears in the meta-data file, not something visible to users.  How strongly do you feel about this?